examples/cfd/tools.py

```diff
@@ -4,27 +4,27 @@
 from matplotlib import pyplot, cm
 
 
-def plot_field(field, xmax=2., ymax=2., zmax=None, view=None, linewidth=0):
+def plot_field(field, xmin=0., xmax=2., ymin=0., ymax=2., zmin=None, zmax=None, view=None, linewidth=0):
     """Utility plotting routine for 2D data
 
     :param field: Numpy array with field data to plot
     :param xmax: (Optional) Length of the x-axis
     :param ymax: (Optional) Length of the y-axis
     :param view: (Optional) View point to intialise
     """
-    x_coord = np.linspace(0, xmax, field.shape[0])
-    y_coord = np.linspace(0, ymax, field.shape[1])
+    x_coord = np.linspace(xmin, xmax, field.shape[0])
+    y_coord = np.linspace(ymin, ymax, field.shape[1])
     fig = pyplot.figure(figsize=(11, 7), dpi=100)
     ax = fig.gca(projection='3d')
     X, Y = np.meshgrid(x_coord, y_coord, indexing='ij')
     ax.plot_surface(X, Y, field[:], cmap=cm.viridis, rstride=1, cstride=1,
                     linewidth=linewidth, antialiased=False)
 
     # Enforce axis measures and set view if given
-    ax.set_xlim(0., xmax)
-    ax.set_ylim(0., ymax)
+    ax.set_xlim(xmin, xmax)
+    ax.set_ylim(ymin, ymax)
     if zmax is not None:
-        ax.set_zlim(1., zmax)
+        ax.set_zlim(zmin, zmax)
```

Comment:
zmin or 0.

not sure why was starting at 1

Reply:
I believe because in all cfd examples the field was initialized to 1, so to avoid seeing an "elevated" field with a non-useful gap. Up to you.
The tutorial plots as they are now look fine, if we change to zmin=0(by default) they would be a bit uglier. In any case it is not like x and y where it is easy for user to know the space extend. I say that we leave it as it always was and maybe some day that tutorials will be revisited we can touch it again.

Reply:
then maybe `zmin or np.min(field)`

Reply:
np.min(field) is the best I think...

Reply:
no `zmin` defaults to  `None` so the correct synthax is `zmin or np.min(field)` so that input is not ignored if not `None`